STUDIO/org.openl.rules.repository.aws/src/org/openl/rules/repository/aws/S3Repository.java

```diff
@@ -111,14 +111,18 @@ public void initialize() {
                 // Possibly don't have permission
                 log.warn("Cannot detect bucket versioning configuration.");
             }
-
-            // Check the connection
-            s3.listObjectsV2(bucketName);
-            list("");
-            monitor = new ChangesMonitor(new S3RevisionGetter(), listenerTimerPeriod);
-        } catch (SdkClientException | IOException e) {
-            throw new IllegalStateException("Failed to initialize a repository", e);
+        } catch (SdkClientException e) {
+            log.warn("Failed to initialize a repository", e);
         }
+
+        monitor = new ChangesMonitor(new S3RevisionGetter(), listenerTimerPeriod);
+    }
+
+    @Override
+    public void validateConnection() throws IOException {
+        // Check the connection
+        s3.listObjectsV2(bucketName);
+        list("");

```

Comment:
Honestly, it's quite expensive operation just for checking the connection.
The step above does the same job. Doesn't it? If the connection is invalid it should throw exception

Reply:
The line above isn't enough: it only checks that the connection is valid and that the bucket exists. But we also need to check that versioning is enabled on the bucket because without it our repository will not work properly.

We can't rely on s3.getBucketVersioningConfiguration(bucketName).getStatus() because we can have no permission for that operation.

But you are right, we don't need two requests to check the connection. I will replace these 2 lines with s3.listVersions(bucketName, "");

Reply:
Looks like listVersions() doesn't work as I expected, so I will stick with getBucketVersioningConfiguration() and take into account that sometimes this operations isn't allowed.